src/MainWindow.cpp

```diff
@@ -4362,71 +4308,83 @@ void MainWindow::on_toolTemplatesLoadAction_triggered()
     p->theProperties->resetValues();
 }
 
-void MainWindow::updateLanguage()
-{
-    if (qtTranslator) {
-        QCoreApplication::removeTranslator(qtTranslator);
-    }
-    if (merkaartorTranslator) {
-        QCoreApplication::removeTranslator(merkaartorTranslator);
-    }
-    QString DefaultLanguage = getDefaultLanguage();
-    if (DefaultLanguage != "-" && DefaultLanguage != "en")
-    {
+QString MainWindow::translationsPath() {
+    return QCoreApplication::applicationDirPath() + "/translations";
+}
+
 #if defined(Q_OS_MAC)
-        QDir resources = QDir(QCoreApplication::applicationDirPath());
-        resources.cdUp();
-        resources.cd("Resources");
+QString MainWindow::macOsTranslationsPath() {
+    QDir resources = QDir(QCoreApplication::applicationDirPath());
+    resources.cdUp();
+    resources.cd("Resources");
+    return resources.absolutePath();
+}
 #endif
-        qtTranslator = new QTranslator;
-        bool retQt;
-        if (g_Merk_Portable) {
-            retQt = qtTranslator->load("qt_" + DefaultLanguage.left(2), QCoreApplication::applicationDirPath() + "/translations");
-        } else {
+
+bool MainWindow::tryLoadQtTranslator(const QString& languagePrefix)
```

Comment:
This function replaces the part of updateLanguage that loaded `qtTranslator` from various paths to avoid the complicated `retQt` boolean and to make the control flow more clear.

This method should be reviewed more carefully. I think the new control flow is equivalent to what happened to `retQt` before, but it should be checked.